### PR TITLE
Fixes #3790- Use WorkManagers to upload contributions

### LINKED ISSUE #3790: Switch to WorkManagers
**Summary:** 

Summarize your issue in one sentence (what goes wrong, what did you expect to happen)
App crashes when upload is somehow initiated from the background.

**Steps to reproduce:** 
Yet to figure out, reported by email

**System logs:**

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.contributions.MainActivity}: java.lang.IllegalStateException: Not allowed to start service Intent { act=fr.free.nrw.commons.upload.upload cmp=fr.free.nrw.commons/.upload.UploadService }: app is in background uid UidRecord{a10f242 u0a101 LAST bg:+10h50m21s78ms idle change:cached procs:1 seq(0,0,0)}
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2951)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086)
	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:4823)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4732)
	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
	at android.app.ClientTransactionHandler.executeTransaction(ClientTransactionHandler.java:55)
	at android.app.ActivityThread.handleRelaunchActivityLocally(ActivityThread.java:4782)
	at android.app.ActivityThread.access$3200(ActivityThread.java:200)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1826)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6718)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { act=fr.free.nrw.commons.upload.upload cmp=fr.free.nrw.commons/.upload.UploadService }: app is in background uid UidRecord{a10f242 u0a101 LAST bg:+10h50m21s78ms idle change:cached procs:1 seq(0,0,0)}
	at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1577)
	at android.app.ContextImpl.startService(ContextImpl.java:1532)
	at android.content.ContextWrapper.startService(ContextWrapper.java:664)
	at fr.free.nrw.commons.contributions.MainActivity.initMain(MainActivity.java:118)
	at fr.free.nrw.commons.contributions.MainActivity.onCreate(MainActivity.java:90)
	at android.app.Activity.performCreate(Activity.java:7144)
	at android.app.Activity.performCreate(Activity.java:7135)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2931)
	... 16 more
```

**Device and Android version:** 
ANDROID_VERSION=9
PHONE_MODEL=Redmi 3S
**Commons app version:** 
APP_VERSION_CODE=709
APP_VERSION_NAME=2.13

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by ashishkumar468**
**Description (required)**

Fixes #3790 

What changes did you make and why?
Use WorkManagers to process upload contributions
** Removed UploadService and Added UploadWorker to process contributions Upload
** Made necessary changes to remove the usages of the Service from the classes
** UI Fixes- Minor changes in the retry and cancel upload icons to give them a clickable area of 48 dp

**Tests performed (required)**

Tested betaDebug on Pixel 4A- API 30
Tested all possible happy cases for the following
* Single file upload
* Multi-File Upload
* Nearby Upload - Wikidata happens
* Limited Connection Mode - ON/OFF
* Test Pause/Resume Uploads

### CODE REVIEW

File: app/build.gradle
Comment (ashishkumar468) [inline app/build.gradle:156]: This is recommended now

File: app/src/main/AndroidManifest.xml
Comment (ashishkumar468) [inline app/src/main/AndroidManifest.xml:138]: Deleted the service

**Comment by neslihanturan** [comment]
Hi @ashishkumar468 , I have made a few tests on this PR. All tests are made on beta debug:

- Upload from gallery:
    - Upload has started as always, however it was very slow.
    - According to the progress bar upload finished.
    - But even progress bar finishes with no error, upload has not made.
    - File is not uploaded and there is no thumbnail on contrib list. Only placeholder. 
- Upload from camera:
    - Upload again was very slow.
    - It failed once, I retried with retry button.
    - The rest is same with the flow for gallery.
  Here are how contrib list was looking like after my these 2 attempt:

 After that I went to master branch. The missing two uploads have been made right after I compiled master branch (since there is a file as "kedi" it is renamed to "kedi4", since there is a file as "flower" it is renamed to "flower2") and they are added to contributions list. However, the items at contributions list from my previous upload attempt (unsuccessful uploads, only placeholder is displayed) stayed remain on the list. Here how contributions list is appeared when I switch to master branch:

**Comment by ashishkumar468** [comment]
Hi @neslihanturan, can you share the logs from when the uploads don't happen

**Comment by neslihanturan** [comment]
Sure, would this one work?
```
21-03-23 18:10:07.528 20595-31791/fr.free.nrw.commons.beta E/ContributionBoundaryCallback$fetchContributions: Failed to fetch contributions: closed
2021-03-23 18:10:07.572 20595-31790/fr.free.nrw.commons.beta E/OkHttpConnectionFactory$UnsuccessfulResponseInterceptor: java.io.IOException: {"error":{"code":"urlparamnormal","info":"Could not normalize image parameters for LastTest-_-_1.jpeg.","docref":"See https://commons.wikimedia.beta.wmflabs.org/w/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list at &lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; for notice of API deprecations and breaking changes."},"servedby":"deployment-mediawiki-07"}
        at fr.free.nrw.commons.OkHttpConnectionFactory$UnsuccessfulResponseInterceptor.intercept(OkHttpConnectionFactory.java:90)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:225)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:254)
        at okhttp3.RealCall.execute(RealCall.java:92)
        at retrofit2.OkHttpCall.execute(OkHttpCall.java:190)
        at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:42)
        at io.reactivex.Observable.subscribe(Observable.java:12267)
        at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:34)
        at io.reactivex.Observable.subscribe(Observable.java:12267)
        at io.reactivex.internal.operators.observable.ObservableSingleSingle.subscribeActual(ObservableSingleSingle.java:35)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleFlatMap.subscribeActual(SingleFlatMap.java:36)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
2021-03-23 18:10:07.572 20595-31790/fr.free.nrw.commons.beta D/OkHttp: <-- 200 https://commons.wikimedia.beta.wmflabs.org/w/api.php?action=query&format=json&formatversion=2&generator=allimages&gaisort=timestamp&gaidir=older&prop=imageinfo&iiprop=url|extmetadata|user&&iiurlwidth=640&iiextmetadatafilter=DateTime|Categories|GPSLatitude|GPSLongitude|ImageDescription|DateTimeOriginal|Artist|LicenseShortName|LicenseUrl&gaiuser=Neslihan%20Test&gailimit=10&gaicontinue=20180701192418|Tst_4.jpg&continue=gaicontinue|| (417ms, unknown-length body)
2021-03-23 18:10:07.574 20595-31790/fr.free.nrw.commons.beta E/ContributionBoundaryCallback$fetchContributions: Failed to fetch contributions: closed
2021-03-23 18:10:07.624 20595-31811/fr.free.nrw.commons.beta D/OkHttp: --> GET https://commons.wikimedia.beta.wmflabs.org/w/api.php?action=query&format=json&formatversion=2&generator=allimages&gaisort=timestamp&gaidir=older&prop=imageinfo&iiprop=url|extmetadata|user&&iiurlwidth=640&iiextmetadatafilter=DateTime|Categories|GPSLatitude|GPSLongitude|ImageDescription|DateTimeOriginal|Artist|LicenseShortName|LicenseUrl&gaiuser=Neslihan%20Test&gailimit=10&gaicontinue=20180701192418|Tst_4.jpg&continue=gaicontinue||
2021-03-23 18:10:07.627 20595-31811/fr.free.nrw.commons.beta D/CookieManager: Domain:commons.wikimedia.beta.wmflabs.org
2021-03-23 18:10:07.717 20595-32008/fr.free.nrw.commons.beta E/OkHttpConnectionFactory$UnsuccessfulResponseInterceptor: java.io.IOException: {"error":{"code":"urlparamnormal","info":"Could not normalize image parameters for LastTest-_-_1.jpeg.","docref":"See https://commons.wikimedia.beta.wmflabs.org/w/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list at &lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; for notice of API deprecations and breaking changes."},"servedby":"deployment-mediawiki-07"}
        at fr.free.nrw.commons.OkHttpConnectionFactory$UnsuccessfulResponseInterceptor.intercept(OkHttpConnectionFactory.java:90)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:225)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:254)
        at okhttp3.RealCall.execute(RealCall.java:92)
        at retrofit2.OkHttpCall.execute(OkHttpCall.java:190)
        at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:42)
        at io.reactivex.Observable.subscribe(Observable.java:12267)
        at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:34)
        at io.reactivex.Observable.subscribe(Observable.java:12267)
        at io.reactivex.internal.operators.observable.ObservableSingleSingle.subscribeActual(ObservableSingleSingle.java:35)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleFlatMap.subscribeActual(SingleFlatMap.java:36)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
        ```

**Comment by ashishkumar468** [comment]
Hi @neslihanturan  Thanks for sharing the logs, I am looking into it

**Comment by ashishkumar468** [comment]
Hi @neslihanturan , I have updated the PR with a fix, verified the upload on Android 10, could you re-verify please?

**Comment by neslihanturan** [comment]
Hey @ashishkumar468 , thanks for the fixes. Now upload succeeded but after select the image I have just uploaded to see media details, it crashed with error:
```
-03-31 22:34:20.084 2178-2178/com.android.systemui D/StatusBar: disable<e i a s b h!r!c s > disable2<q i n >
2021-03-31 22:34:20.085 1133-1288/system_process W/UsageStatsService: Event reported without a package name, eventType:18
2021-03-31 22:34:20.089 2178-2178/com.android.systemui D/KeyguardUpdateMonitor: onKeyguardVisibilityChanged(false)
2021-03-31 22:34:20.397 2178-2178/com.android.systemui D/ImageWallpaper: wallpaper visibility changes to: false
2021-03-31 22:34:20.584 2149-8607/fr.free.nrw.commons.beta D/OkHttp: <-- 200 https://commons.wikimedia.beta.wmflabs.org/w/api.php?format=json&formatversion=2&errorformat=plaintext&action=query&meta=notifications&notformat=model&notlimit=max&notwikis=wikidatawiki%7Ccommonswiki%7Cenwiki&notfilter=%21read (544ms, unknown-length body)
2021-03-31 22:34:20.615 2149-2149/fr.free.nrw.commons.beta D/ContributionsFragment: Number of notifications is 8
2021-03-31 22:34:21.086 2178-2406/com.android.systemui D/ImageWallpaper: finishRendering, preserve=false
2021-03-31 22:34:21.469 2149-2149/fr.free.nrw.commons.beta D/AndroidRuntime: Shutting down VM
    
    
    --------- beginning of crash
2021-03-31 22:34:21.471 2149-2149/fr.free.nrw.commons.beta E/AndroidRuntime: FATAL EXCEPTION: main
    Process: fr.free.nrw.commons.beta, PID: 2149
    android.database.sqlite.SQLiteException: Can't downgrade database from version 15 to 14
        at android.database.sqlite.SQLiteOpenHelper.onDowngrade(SQLiteOpenHelper.java:541)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:415)
        at android.database.sqlite.SQLiteOpenHelper.getReadableDatabase(SQLiteOpenHelper.java:341)
        at fr.free.nrw.commons.bookmarks.pictures.BookmarkPicturesContentProvider.query(BookmarkPicturesContentProvider.java:61)
        at android.content.ContentProvider.query(ContentProvider.java:1214)
        at android.content.ContentProvider.query(ContentProvider.java:1307)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:267)
        at android.content.ContentProviderClient.query(ContentProviderClient.java:195)
        at android.content.ContentProviderClient.query(ContentProviderClient.java:177)
        at android.content.ContentProviderClient.query(ContentProviderClient.java:167)
        at fr.free.nrw.commons.bookmarks.pictures.BookmarkPicturesDao.findBookmark(BookmarkPicturesDao.java:132)
        at fr.free.nrw.commons.media.MediaDetailPagerFragment.updateBookmarkState(MediaDetailPagerFragment.java:318)
        at fr.free.nrw.commons.media.MediaDetailPagerFragment.onCreateOptionsMenu(MediaDetailPagerFragment.java:273)
        at androidx.fragment.app.Fragment.performCreateOptionsMenu(Fragment.java:2808)
        at androidx.fragment.app.FragmentManager.dispatchCreateOptionsMenu(FragmentManager.java:2664)
        at androidx.fragment.app.Fragment.performCreateOptionsMenu(Fragment.java:2810)
        at androidx.fragment.app.FragmentManager.dispatchCreateOptionsMenu(FragmentManager.java:2664)
        at androidx.fragment.app.FragmentController.dispatchCreateOptionsMenu(FragmentController.java:387)
        at androidx.fragment.app.FragmentActivity.onCreatePanelMenu(FragmentActivity.java:325)
        at androidx.appcompat.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:94)
        at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.onCreatePanelMenu(AppCompatDelegateImpl.java:2830)
        at androidx.appcompat.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:94)
        at androidx.appcompat.app.ToolbarActionBar.populateOptionsMenu(ToolbarActionBar.java:455)
        at androidx.appcompat.app.ToolbarActionBar$1.run(ToolbarActionBar.java:56)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:967)
        at android.view.Choreographer.doCallbacks(Choreographer.java:791)
        at android.view.Choreographer.doFrame(Choreographer.java:722)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:952)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)
2021-03-31 22:34:21.472 2149-2149/fr.free.nrw.commons.beta E/ACRA: ACRA caught a SQLiteException for fr.free.nrw.commons.beta
    android.database.sqlite.SQLiteException: Can't downgrade database from version 15 to 14
        at android.database.sqlite.SQLiteOpenHelper.onDowngrade(SQLiteOpenHelper.java:541)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:415)
        at android.database.sqlite.SQLiteOpenHelper.getReadableDatabase(SQLiteOpenHelper.java:341)
        at fr.free.nrw.commons.bookmarks.pictures.BookmarkPicturesContentProvider.query(BookmarkPicturesContentProvider.java:61)
        at android.content.ContentProvider.query(ContentProvider.java:1214)
        at android.content.ContentProvider.query(ContentProvider.java:1307)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:267)
        at android.content.ContentProviderClient.query(ContentProviderClient.java:195)
        at android.content.ContentProviderClient.query(ContentProviderClient.java:177)
        at android.content.ContentProviderClient.query(ContentProviderClient.java:167)
        at fr.free.nrw.commons.bookmarks.pictures.BookmarkPicturesDao.findBookmark(BookmarkPicturesDao.java:132)
        at fr.free.nrw.commons.media.MediaDetailPagerFragment.updateBookmarkState(MediaDetailPagerFragment.java:318)
        at fr.free.nrw.commons.media.MediaDetailPagerFragment.onCreateOptionsMenu(MediaDetailPagerFragment.java:273)
        at androidx.fragment.app.Fragment.performCreateOptionsMenu(Fragment.java:2808)
        at androidx.fragment.app.FragmentManager.dispatchCreateOptionsMenu(FragmentManager.java:2664)
        at androidx.fragment.app.Fragment.performCreateOptionsMenu(Fragment.java:2810)
        at androidx.fragment.app.FragmentManager.dispatchCreateOptionsMenu(FragmentManager.java:2664)
        at androidx.fragment.app.FragmentController.dispatchCreateOptionsMenu(FragmentController.java:387)
        at androidx.fragment.app.FragmentActivity.onCreatePanelMenu(FragmentActivity.java:325)
        at androidx.appcompat.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:94)
        at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.onCreatePanelMenu(AppCompatDelegateImpl.java:2830)
        at androidx.appcompat.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:94)
        at androidx.appcompat.app.ToolbarActionBar.populateOptionsMenu(ToolbarActionBar.java:455)
        at androidx.appcompat.app.ToolbarActionBar$1.run(ToolbarActionBar.java:56)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:967)
        at android.view.Choreographer.doCallbacks(Choreographer.java:791)
        at android.view.Choreographer.doFrame(Choreographer.java:722)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:952)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)
2021-03-31 22:34:21.544 1133-3058/system_process I/ActivityTaskManager: START u0 {flg=0x10000000 cmp=fr.free.nrw.commons.beta/org.acra.dialog.CrashReportDialog (has extras)} from uid 10154
2021-03-31 22:34:21.553 2149-2149/fr.free.nrw.commons.beta I/Process: Sending signal. PID: 2149 SIG: 9
2021-03-31 22:34:21.682 1133-1989/system_process W/InputDispatcher: channel '280d321 fr.free.nrw.commons.beta/fr.free.nrw.commons.contributions.MainActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
2021-03-31 22:34:21.682 1133-1989/system_process E/InputDispatcher: channel '280d321 fr.free.nrw.commons.beta/fr.free.nrw.commons.contributions.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2021-03-31 22:34:21.684 1133-8674/system_process I/ActivityManager: Process fr.free.nrw.commons.beta (pid 2149) has died: fore TOP 
2021-03-31 22:34:21.689 1133-3058/system_process I/WindowManager: WIN DEATH: Windo
```

**Review by neslihanturan — APPROVED** [review]
Thanks @ashishkumar468 , it works. Will merge right after conflicts are fixed :)

**Comment by ashishkumar468** [comment]
Hi @neslihanturan , @misaochan - I have updated the PR, synced it with the master branch to resolve merge conflicts. Please have a look.

**Review by neslihanturan — APPROVED** [review]
Thanks a lot @ashishkumar468 :)

**Pull request merged**
